content/gocourse-2019-05-melbourne.slide

```diff
@@ -816,49 +816,56 @@ Embedding types introduces the possiblity for naming conflicts. Go follows some
 .caption J. D. Frazer "Illiad", User Friendly - [[https://en.wikipedia.org/wiki/User_Friendly#Crud_Puppy][Crud Puppy]]
 
 
-* Errors (1/3)
+* Errors in Go
 
-"Errors are values."  - [[https://go-proverbs.github.io/][Rob Pike]]
-
-The `error` type is a built-in interface:
+- `error` is a builtin interface ([[https://golang.org/src/builtin/builtin.go?s=11246:11286#L251][source: error]])
+- Check errors explicitly
+- The [[https://golang.org/pkg/errors/][`errors`]] package has functions for creating new error values
+- Using `errors.New()`:
 
-	type error interface {
-	    Error() string
+	func Sqrt(f float64) (float64, error) {
+		if f < 0 {
+			return 0, errors.New("math: square root of negative number")
+		}
+		// implementation
+	}
+- Alternatively using `fmt.Errorf`:
+	if f < 0 {
+		return 0, fmt.Errorf("math: square root of negative number %g", f)
 	}
 
-By convention, functions often return `error` as the second/last return value.
-The calling code should handle errors by testing whether the error equals nil:
+* Errors in Go (continued)
```

Comment:
I think you do need somewhere what the error interface looks like that you are implmenting, it's pretty short:
```go
type error interface {
	    Error() string
}
```

Reply:
Done. Added it in the first slide. The (source: error) was a link to that. I've also put it up on the slide